frontend/components/siteContainer.js

```diff
@@ -1,53 +1,95 @@
 import React from 'react';
-import { Link } from 'react-router';
-
-import { replaceHistory } from '../actions/routeActions';
-import siteActions from '../actions/siteActions';
+import PropTypes from 'prop-types';
 
 import SideNav from './site/SideNav/sideNav';
 import PagesHeader from './site/pagesHeader';
 import AlertBanner from './alertBanner';
+import LoadingIndicator from './loadingIndicator';
 
 const propTypes = {
-  storeState: React.PropTypes.object,
-  params: React.PropTypes.object //{id, branch, splat, fileName}
+  params: PropTypes.shape({
+    id: PropTypes.string,
+    branch: PropTypes.string,
+    fileName: PropTypes.string,
+  }).isRequired,
+  storeState: PropTypes.shape({
+    sites: PropTypes.object,
+    builds: PropTypes.object,
+    buildLogs: PropTypes.object,
+    publishedBranches: PropTypes.object,
+    publishedFiles: PropTypes.object,
+    githubBranches: PropTypes.object,
+    alert: PropTypes.shape({
+      message: PropTypes.string,
+      status: PropTypes.string,
+    }),
+  }),
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }).isRequired,
+  children: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.node),
+    PropTypes.node,
+  ]),
 };
 
-class SiteContainer extends React.Component {
-  constructor(props) {
-    super(props);
-  }
+const defaultProps = {
+  children: null,
+  storeState: null,
+};
 
+class SiteContainer extends React.Component {
   getPageTitle(pathname) {
     return pathname.split('/').pop();
   }
 
   getCurrentSite(sitesState, siteId) {
     if (sitesState.isLoading) {
-      return null
+      return null;
     }
 
-    return sitesState.data.find((site) => {
-      // force type coersion
-      return site.id == siteId;
-    });
+    return sitesState.data.find(site => site.id === parseInt(siteId, 10));
   }
 
-  render () {
+  render() {
     const { storeState, children, params, location } = this.props;
 
-    const site = this.getCurrentSite(storeState.sites, params.id)
-    const builds = storeState.builds
-    const buildLogs = storeState.buildLogs
-    const publishedBranches = storeState.publishedBranches
-    const publishedFiles = storeState.publishedFiles
-    const githubBranches = storeState.githubBranches
-    const childConfigs = { site, builds, buildLogs, publishedBranches, publishedFiles, githubBranches }
+    const site = this.getCurrentSite(storeState.sites, params.id);
+    const builds = storeState.builds;
+    const buildLogs = storeState.buildLogs;
+    const publishedBranches = storeState.publishedBranches;
+    const publishedFiles = storeState.publishedFiles;
+    const githubBranches = storeState.githubBranches;
+    const childConfigs = {
+      site,
+      builds,
+      buildLogs,
+      publishedBranches,
+      publishedFiles,
+      githubBranches,
+    };
 
     const pageTitle = this.getPageTitle(location.pathname);
 
     if (!site) {
-      return null;
+      if (storeState.sites.isLoading) {
```

Comment:
This works, but the nested conditionals make me a little antsy. Any way we could extract the missing site case into another function or otherwise flatten out the conditional?